### ISSUE TITLE
Keras compatability -- easy addition?

### ISSUE THREAD

**Issue by jmhessel** [JianGoForIt/YellowFin#1]
Hi! Thanks for posting this code. I thought that I would give YF a try as a drop-in optimizer. Currently, I am using [Keras](keras.io), and I was able to modify your code to run on Keras models by doing the following:

- Adding a `compute_gradients` standalone method
- Adding a few checks for gradients being None in `apply_gradients` and `after_apply`
- Wrapping the `YFOptimizer` object in a Keras [TFOptimizer](https://keras.io/optimizers/#tfoptimizer)

However, while it runs and my loss goes down -- I am not 100% I did everything properly. Do you think you might consider adding this support?

**Comment by Zehaos** [comment]
Hi @jmhessel, 

I'm now going through what you have done, I wanna to know wheather you have trained a reasonable model？ and what dataset are you training on?

Best,
Zehao

**Comment by Zehaos** [comment]
@jmhessel  I have validated it on mnist dataset, got 97.64% accuracy.

**Comment by JianGoForIt** [comment]
Hi @Zehaos @jmhessel,

Thanks for helping us validating and integrating.

@jmhessel 
For the three points, I think we can add here the first two points. For the last one, it would be the best practice to request Keras team to integrate. We could try to reach out. What do you think?

Cheers,

Jian

**Comment by JianGoForIt** [comment]
@jmhessel Could you please do a pull request with your modification for the first two points? 

Thanks.

**Comment by jmhessel** [comment]
@JianGoForIt Sure -- I can do a quick PR (I actually didn't even bother saving the code because the modifications were so minor, but it will be easy to re-do once I get the chance in the next few days). With respect to point 3, I don't think you'll need to get in contact with the keras folks -- when I say that the YFOptimizer is wrapped in the TFOptimizer, I mean that it is done programmatically (i.e., it's not a subclass if TFOptimizer or anything, you simply write `opt=TFOptimizer(YFOptimizer())` which is pretty clean, as is.

**Comment by JianGoForIt** [comment]
@jmhessel Ha, got your point right this time. That sounds great! I will merge your PR as soon as it is done.

**Comment by JianGoForIt** [comment]
Keras support added using commits for pull request here https://github.com/JianGoForIt/YellowFin/commit/0949dfe2e674e1b83bdd06bbc21cf9b2d920ae1f
